pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -160,35 +160,40 @@ func (r *AtlasProjectReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	if !owner {
 		result = workflow.Terminate(
 			workflow.AtlasDeletionProtection,
-			"unable to reconcile project: it already exists in Atlas, it was not previously managed by the operator, and the deletion protection is enabled.",
+			"unable to reconcile Project due to deletion protection being enabled. see https://dochub.mongodb.org/core/ako-deletion-protection for further information",
 		)
 		workflowCtx.SetConditionFromResult(status.ProjectReadyType, result)
 		log.Error(result.GetMessage())
 
 		return result.ReconcileResult(), nil
 	}
 
-	err = customresource.ApplyLastConfigApplied(ctx, project, r.Client)
-	if err != nil {
-		result = workflow.Terminate(workflow.Internal, err.Error())
-		workflowCtx.SetConditionFromResult(status.ProjectReadyType, result)
-		log.Error(result.GetMessage())
-
-		return result.ReconcileResult(), nil
-	}
-
-	var projectID string
-	if projectID, result = r.ensureProjectExists(workflowCtx, project); !result.IsOk() {
+	projectID, result := r.ensureProjectExists(workflowCtx, project)
+	if !result.IsOk() {
 		setCondition(workflowCtx, status.ProjectReadyType, result)
 		return result.ReconcileResult(), nil
 	}
+
 	workflowCtx.EnsureStatusOption(status.AtlasProjectIDOption(projectID))
 
-	if result := r.ensureDeletionFinalizer(ctx, workflowCtx, atlasClient, project); !result.IsOk() {
+	if result = r.ensureDeletionFinalizer(ctx, workflowCtx, atlasClient, project); !result.IsOk() {
 		setCondition(workflowCtx, status.ProjectReadyType, result)
 		return result.ReconcileResult(), nil
 	}
 
+	if project.ID() == "" {
+		err = customresource.ApplyLastConfigApplied(ctx, project, r.Client)
+		if err != nil {
```

Comment:
Why are we applying last config here? seems like an early return?

Reply:
In order to properly check subresources, the last config annotation must be present, if a new CR is added we first sync the project itself and re-queue to sync the subresources